plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/CdfFileEntry.java

```diff
@@ -35,6 +40,18 @@ public CdfFileEntry(
     {
         this.path = path;
         this.partitionValues = partitionValues;
+        this.canonicalPartitionValues = partitionValues.entrySet().stream()
+                .collect(toImmutableMap(
+                        // canonicalize partition keys to lowercase so they match column names used in DeltaLakeColumnHandle
+                        entry -> canonicalizeColumnName(entry.getKey()),
+                        entry -> {
+                            String value = entry.getValue();
+                            if (value == null || value.isEmpty()) {
+                                // For VARCHAR based partitions null and "" are treated the same
```

Comment:
Should this logic (and in particular this comment) be a shared method reused between this class and AddFileEntry ?

Reply:
sure, I will add that